core/client/fs/src/main/java/alluxio/client/file/options/OutStreamOptions.java

```diff
@@ -130,7 +131,11 @@ private OutStreamOptions(ClientContext context, AlluxioConfiguration alluxioConf
     mWriteTier = alluxioConf.getInt(PropertyKey.USER_FILE_WRITE_TIER_DEFAULT);
     mWriteType = alluxioConf.getEnum(PropertyKey.USER_FILE_WRITE_TYPE_DEFAULT, WriteType.class);
     try {
-      mOwner = context.getUserState().getUser().getName();
+      mOwner = AuthenticationUserUtils.getImpersonationUser(
+          context.getUserState().getSubject(), alluxioConf);
+      if (mOwner == null) {
```

Comment:
What if `mOwner` is not null while `AuthenticationUserUtils.getImpersonationUser()` returns you a user? I think we should output a WARNING log for the unexpected. Also I think it's worth logging in INFO or DEBUG when you are setting the impersonated user. If people run into impersonation issues they can find a trace.